src/bootstrap/defaults/bootstrap.ferrocene-dist.toml

```diff
@@ -170,6 +170,10 @@ channel = "auto-detect"
 
 
 [target]
+# We default to the self-contained lld linker on Linux GNU targets.
+x86_64-unknown-linux-gnu.default-linker-linux-override = 'self-contained-lld-cc'
+aarch-unknown-linux-gnu.default-linker-linux-override = 'self-contained-lld-cc'
```

Comment:
For some reason this doesn't seem to be sufficient? I'm rebasing and cleaning up the musl issues to isolate this.